File: src/money_ops.js
	var MoneyOps=function (){
}

MoneyOps.add = function(m1,m2){

	if(m1 === null && m2 === null){
		throw new MoneyNullValueOrNullCurrency();
	}else if(m1.getValue()<=0 ||m2.getValue()<=0)
		throw new moneyInforEqualZero();
	else if(m1.getCurrency() !== m2.getCurrency())
		throw new DevisesIncompatibleExc(m1,m2);
	else
		return new money(m1.getValue()+m2.getValue(),m1.getCurrency());

}

MoneyOps.sus = function(m1,m2){

	if(m1 === null || m2 === null){
		throw new MoneyNullValueOrNullCurrency();
	}else if(m1.getValue() < m2.getValue())
		throw new ErrorValue();
	else if(m1.getValue()<=0 ||m2.getValue()<=0)
		throw new moneyInforEqualZero();
	else if(m1.getCurrency() !== m2.getCurrency())
		throw new DevisesIncompatibleExc(m1,m2);
	else
		return new money(m1.getValue()-m2.getValue(),m1.getCurrency());


}

File: Money/src/fr/ulille1/fil/odeva/MoneyOps.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package fr.ulille1.fil.odeva;

/**
 *
 * @author marius
 */
public class MoneyOps {
  private static MoneyFactory mf = MoneyFactory.getDefaultFactory();

  /**
   * Add moneys having the same currency
  */
  
  public static void setMoneyFactory(MoneyFactory injectedMF){
	  mf = injectedMF;
  }
  public static Money simpleAdd(Money m1, Money m2) throws NullPointerException,IncompatibleCurrencyException, UnexistingCurrencyException
  {
	  if(m1.getCurrency()!=m2.getCurrency())
		  throw new IncompatibleCurrencyException(m1.getCurrency(), m2.getCurrency());
	  
    Money m=mf.createMoney(m1.getValue()+m2.getValue(),m1.getCurrency());
    return m;
  }
}
